dev_guide/application_lifecycle/promoting_applications.adoc

```diff
@@ -0,0 +1,781 @@
++
+----
+$ oc tag <project_for_stage_N>/<imagestream_name_for_stage_N>:<tag_for_stage_N> <project_for_stage_N+1>/<imagestream_name_for_stage_N+1>:<tag_for_stage_N+1>
+----
+
+.. If the registry is not shared, you can leverage the access tokens for each of
+your promotion pipeline registries as you log into both the source and
+destination registries, pulling, tagging, and pushing your application images
+accordingly:
+
+... Log in to the source environment registry:
++
+----
+$ docker login -u <username> -e <any_email_address> -p <token_value> <src_env_registry_ip>:<port>
+----
+
+... Pull your application's image:
++
+----
+$ docker pull <src_env_registry_ip>:<port>/<namespace>/<image name>:<tag>
+----
+
+... Tag your application's image to the destination registry's location, updating
+namespace, name, and tag as needed to conform to the destination staging
+environment:
++
+----
+$ docker tag <src_env_registry_ip>:<port>/<namespace>/<image name>:<tag> <dest_env_registry_ip>:<port>/<namespace>/<image name>:<tag>
+----
+
+... Log into the destination staging environment registry:
++
+----
+$ docker login -u <username> -e <any_email_address> -p <token_value> <dest_env_registry_ip>:<port>
+----
+
+... Push image to destination:
++
+----
+$ docker push <dest_env_registry_ip>:<port>/<namespace>/<image name>:<tag>
+----
++
+[TIP]
+To automatically import new versions of an image from an external registry, the
+`oc tag` command has a `--scheduled` option. If used, the image the
+`ImageStreamTag` references will be periodically pulled from the registry
+hosting the image.
+
+. Next, there are the cases where the evolution of your application necessitates
+fundamental changes to your API objects or additions and deletions from the set
+of API objects that make up the application. When such evolution in your
+application's API objects occurs, the {product-title} CLI provides a broad range
+of options to transfer to changes from one staging environment to the next.
+
+.. Start in the same fashion as you did when you initially set up your promotion
+pipeline:
++
+----
+$ oc login <source_environment>
+$ oc project <source_project>
+$ oc export dc,is,svc,route,secret,sa -l template=<application_template_name> -o yaml > export.yaml
+$ oc login <target_environment>
+$ oc <target_project>
+----
+
+.. Rather than simply creating the resources in the new environment, update them. You can do this a few different ways:
+
+... The more conservative approach is to leverage `oc apply` and merge the new
+changes to each API object in the target environment. In doing so, you can
+`--dry-run=true` option and examine the resulting objects prior to actually
+changing the objects:
++
+----
+$ oc apply -f export.yaml --dry-run=true
+----
++
+If satisfied, run it for real:
++
+----
+$ oc apply -f export.yaml
+----
++
+The `apply` command optionally takes additional arguments that help with more
+complicated scenarios. See the command line help for the `oc` command and
```

Comment:
Include a link to the CLI reference here and what should be run in the command line to get help?